proposals/p4075.md

```diff
@@ -0,0 +1,129 @@
+# Change operator precedence
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/4075)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Abstract](#abstract)
+-   [Problem](#problem)
+-   [Background](#background)
+-   [Proposal](#proposal)
+-   [Details](#details)
+-   [Rationale](#rationale)
+-   [Alternatives considered](#alternatives-considered)
+    -   [`as` and `where` could be peers of `if`...`then`...`else`](#as-and-where-could-be-peers-of-ifthenelse)
+    -   [Make `T as I where R` mean `T as (I where R)`](#make-t-as-i-where-r-mean-t-as-i-where-r)
+    -   [Make fewer changes](#make-fewer-changes)
+
+<!-- tocstop -->
+
+## Abstract
+
+Update the operator precedence to achieve a few goals:
+
+-   Form operators into groups which behave similarly
+-   Make the group of operators ("top-level operators") that capture everything
+    to the right, like `if`...`then`...`else`, behave similarly to the left, so
+    that rearranging expressions won't change how they group.
+-   Add the `where` operator, used to specify constraints on facet types, to the
+    precedence chart, to define how it interacts with other operators.
+-   Make the operator precedence diagram prettier, so that it eventually can be
+    made into a poster that Carbon programmers can hang on their walls.
+
+## Problem
+
+The `where` operator is particularly tricky:
+
+-   It is used in an `impl` declaration to specify the values of associated
+    constants (such as associated types). In that context, `impl T as I where R`
+    is interpreted conceptually as `impl T as (I where R)`. It would be nice if
+    `T as I where R` would mean the same thing in other contexts.
+-   The `where` operator will frequently be used with the binary `&` operator,
+    since that is how facet types are combined. It is desirable that
+    `I & J where R` be interpreted as `(I & J) where R`. This is expected to be
```

Comment:
Likewise here: considered in isolation, we probably prefer `(I & J) where R` > invalid > `I & (I where R)`.

Reply:
Done.